### PR TITLE
Add squeezenet to CI build

### PR DESCRIPTION
**Pull request by snnn**
1. Add the squeezenet example to CI build
2. Validate all C/C++ examples with the latest onnxruntime nightly build. (Something is better than nothing)

### CODE REVIEW

File: c_cxx/squeezenet/main.cpp [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #include <assert.h>
 #include <vector>
 #include <onnxruntime_cxx_api.h>
+
+#ifdef HAVE_TENSORRT_PROVIDER_FACTORY_H
 #include <tensorrt_provider_factory.h>
 #include <tensorrt_provider_options.h>
```
Comment (chilo-ms) [inline c_cxx/squeezenet/main.cpp:11]: Quick question here.
In line 41, the OrtTensorRTProviderOptionsV2 is defined in tensorrt_provider_options.h
It seems line 41 also should be in #ifdef HAVE_TENSORRT_PROVIDER_FACTORY_H ?

Comment (snnn) [reply c_cxx/squeezenet/main.cpp:11]: Interesting. Why it could work now?

Comment (snnn) [reply c_cxx/squeezenet/main.cpp:11]: SessionOptionsAppendExecutionProvider_TensorRT_V2 is declared in onnxruntime_c_api.h, it works similar. onnxruntime_c_api.h doesn't include tensorrt_provider_options.h.  So I think it is ok.

Comment (snnn) [reply c_cxx/squeezenet/main.cpp:11]: Oh, in onnxruntime_c_api.h we have "ORT_RUNTIME_CLASS(TensorRTProviderOptionsV2);" . So it is enough.

**Comment by snnn** [comment]
@chilo-ms , there are a few warnings, I hope you can help fix: https://github.com/microsoft/onnxruntime-inference-examples/runs/5559177059?check_suite_focus=true

**Comment by chilo-ms** [comment]
> @chilo-ms , there are a few warnings, I hope you can help fix: https://github.com/microsoft/onnxruntime-inference-examples/runs/5559177059?check_suite_focus=true

Okay, I will fix the warnings.

**Review by chilo-ms — APPROVED** [review]

**Pull request merged**